app/objects/c_operation.py

```diff
@@ -187,9 +187,10 @@ def update_untrusted_agents(self, agent):
 
     async def all_facts(self):
         knowledge_svc_handle = BaseService.get_service('knowledge_svc')
+        data_svc_handle = BaseService.get_service('data_svc')
         seeded_facts = []
         if self.source:
-            seeded_facts = await knowledge_svc_handle.get_facts(criteria=dict(source=self.source.id))
+            seeded_facts = await data_svc_handle.get_facts_from_source(self.source.id)
```

Comment:
@clenk  How do you feel about this?

Im given pause as `all_facts()` is called a lot in code base. However, this change seems to be equivalent as only getting facts for the given source.

Reply:
Apart from fixing the issues mentioned, what does change is that modifying the base fact source will directly be reflected on the seeded_facts part, even during an operation. But is that an expected behavior, or is a simple doc update to warn users enough?
I can't think of/have found anything else that could be troublesome.